### PR TITLE
[PIN-112] Display reward pinnies, update cached pinnies

### PR DESCRIPTION
**Pull request by skovati**
Pretty self explanatory tbh

### CODE REVIEW

File: app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java [hunk 1]
```diff
@@ -393,17 +392,27 @@ public Task<HttpsCallableResult> findPin(String pid, Location location) {
...
+                    String broadLocationName = (String) result.get("broadLocationName");
+                    String nearbyLocationName = (String) result.get("nearbyLocationName");
+                    Long reward = Long.parseLong(result.getOrDefault("reward", 0L).toString());
+                    Log.d(TAG, String.format("Got reward for pin: %d", reward));
+
+                    activity.add(new ActivityItem(auth.getUid(), pid, ActivityItem.ActivityType.FIND,
+                            broadLocationName, nearbyLocationName));
+                    foundPinMetadata.add(new PinMetadata(pid, broadLocationName, nearbyLocationName));
+
+                    return reward;
```
Comment (ckleinhans) [inline app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java:411]: what if we just update the cached pinnies in this `continueWith` clause so we don't need to create another public method that exposes  modifying the pinnie count?

Comment (skovati) [reply app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java:411]: Good call

Comment (skovati) [reply app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java]: Change has been made

**Review by ckleinhans — APPROVED** [review]
👍

**Pull request merged**